doc/releases/changelog-dev.md

```diff
@@ -72,6 +72,7 @@
 
 * Added new error tracking and propagation functionality. 
   [(#5115)](https://github.com/PennyLaneAI/pennylane/pull/5115)
+  [(#5121)](https://github.com/PennyLaneAI/pennylane/pull/5121)
```

Comment:
I think it would be helpful, to keep track of the work, if we had a title (description) for each PR.

Reply:
Hey, this is a good point. Since we will have multiple PRs associated with this feature, our strategy right now is to group them all into a single changelog entry and provide a more wholistic explanation of the functionality (for example, see the entry `Upgrade Pauli arithmetic:` under the improvements section of the changelog). 

Regardless, before the release @trbromley usually goes through the changelog and re-organizes it as needed. So we can change it then if this approach is not as clear 👍🏼

Reply:
Yep, having an entry for each PR is less important than having a single well explained entry with examples. If this happens during the PR stage then great, but otherwise @DSGuala and myself will have a revision closer to release.